pkg/helpers/tokencmd/negotiator_gssapi.go

```diff
@@ -4,195 +4,91 @@
 package tokencmd
 
 import (
-	"errors"
-	"runtime"
-	"sync"
-	"time"
+	"os"
 
-	"github.com/apcera/gssapi"
-	"k8s.io/klog/v2"
+	krb5client "github.com/jcmturner/gokrb5/v8/client"
+	krb5conf "github.com/jcmturner/gokrb5/v8/config"
+	krb5kt "github.com/jcmturner/gokrb5/v8/keytab"
+)
 
-	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+const (
+	krb5Config       = "/etc/krb5.conf"
+	krb5ConfigEnvKey = "KRB5_CONFIG"
+	krb5Keytab       = "/etc/krb5.keytab"
+	krb5KeytabEnvKey = "KRB5_CLIENT_KTNAME"
 )
 
 func GSSAPIEnabled() bool {
 	return true
 }
 
 type gssapiNegotiator struct {
-	// handle to a loaded gssapi lib
-	lib *gssapi.Lib
-	// error seen when loading the lib
-	loadError error
-	// lock to make sure we only load it once
-	loadOnce sync.Once
-
-	// service name of the server we are negotiating against
-	name *gssapi.Name
-	// security context holding the state of the negotiation
-	ctx *gssapi.CtxId
-	// flags indicating what options we want for the negotiation
-	// TODO: surface option for mutual authentication, e.g. gssapi.GSS_C_MUTUAL_FLAG
-	flags uint32
-
 	// principalName contains the name of the principal desired by the user, if specified.
 	principalName string
-	// cred holds the credentials obtained for the specified principalName.
-	// if no principalName is specified, GSS_C_NO_CREDENTIAL is used.
-	cred *gssapi.CredId
 
 	// track whether the last response from InitSecContext was GSS_S_COMPLETE
 	complete bool
+
+	client *krb5client.Client
 }
 
 func NewGSSAPINegotiator(principalName string) Negotiator {
 	return &gssapiNegotiator{principalName: principalName}
 }
 
 func (g *gssapiNegotiator) Load() error {
-	_, err := g.loadLib()
-	return err
+	return nil
 }
 
 func (g *gssapiNegotiator) InitSecContext(requestURL string, challengeToken []byte) (tokenToSend []byte, err error) {
```

Comment:
@LorbusChris we are not using `challengeToken` now. I checked it and it seems that imported library handles that automatically, could you please confirm me?

Reply:
I'm no expert here, but that would be my impression tool. These changes will have to be tested against a real cluster that uses kerberos auth for verification.